src/MOI_wrapper.jl

```diff
@@ -2488,3 +2488,61 @@ function MOI.get(
     end
     return get_intattrelement(model.inner, "IISQConstr", _info(model, index).row) > 0
 end
+
+###
+### Constraint attributes
+###
+
+struct ConstraintAttribute <: MOI.AbstractConstraintAttribute
+    name::String
+end
+
+const constr_attr_type =
+    Dict( "Lazy" => Int, "DStart" => Real, "IISConstr" => Int,
+          "SARHSLow" => Real, "SARHSUp" => Real, "FarkasDual" => Real,
+          "Sense" => Char, "RHS" => Real, "ConstrName" => String, "Pi" => Real,
+          "Slack" => Real, " CBasis" => Int )
+const getter_for_type = Dict( Int => get_intattrelement,
+                              Real => get_dblattrelement,
```

Comment:
`set_intattrelement!()` takes an `Integer` and `set_dblattrelement!` takes a `Real`, shouldn't `MOI.set()` do the same?

Reply:
> Did you consider
> 
> ```julia
> struct ConstraintAttribute{T}
>     name::String
> end
> function ConstraintAttribute(name::String)
>     return ConstraintAttribute{CONSTR_ATTR_TYPE[name]}(name)
> end
> 
> function MOI.set(model::Optimizer, attr::ConstraintAttribute{Int}, c::MOI.ConstraintIndex, val::Int)
>     set_intattrelement!(model.inner, attr.name, _info(model, ci).row, val)
>     _require_update(model)
> end
> ```

This looks more Juliaesque but it handles user errors poorly:
```Julia
MOI.set(db, Gurobi.ConstraintAttribute("Lazy"), index(c), 3.0)
```
Output:
`ERROR: MathOptInterface.UnsupportedAttribute{Gurobi.ConstraintAttribute{Int64}}: Attribute Gurobi.ConstraintAttribute{Int64}("Lazy") is not supported by the model.`

The error message is misleading since the attribute is supported by the model. The error here is that the value passed is of the wrong type.